nodejs/eks/cluster.ts

```diff
@@ -140,6 +184,26 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
       }, { parent: parent })).role;
     }
 
+    // Create the EKS cluster IAM Role Provider
+    let creationRole: pulumi.Output<aws.iam.Role>;
+    if (args.creationRole) {
+        creationRole = pulumi.output(args.creationRole);
+    } else {
+        creationRole = createEntityRole(name, parent);
+    }
+    const eksCreationRoleArn = creationRole.apply(role => role.arn);
+    const eksClusterProvider = new aws.Provider(`${name}-eksClusterCreatorEntity`, {
+        assumeRole: {
+            roleArn: eksCreationRoleArn.apply(async(arn) => {
+                // wait 30 seconds to assume the IAM Role https://github.com/pulumi/pulumi-aws/issues/673
+                if (!pulumi.runtime.isDryRun()) {
+                    await new Promise(resolve => setTimeout(resolve, 30 * 1000));
+                }
+                return arn;
+            }),
+        },
+    }, { parent: parent });
```

Comment:
I’ll admit I still don’t exactly follow why we have to create a new provider. Does EKS implicitly take the role that was used to create it and bake it in somewhere?  That would be really weird. Is there not a way to just pass it the role you want it to associate with IAM access?

Reply:
How does this handle region and other provide configuration?  I’m not sure there is any automatic propagation of such things from the parents provider down to a newly constructed provider instance. (Perhaps there should be?)

Reply:
> Does EKS implicitly take the role that was used to create it and bake it in somewhere? That would be really weird. Is there not a way to just pass it the role you want it to associate with IAM access?

Yes, the role is inferred and there isn't a way to specify the it as a a param when creating the EKS cluster.

From the AWS [docs](https://docs.aws.amazon.com/eks/latest/userguide/add-user-role.html): `When you create an Amazon EKS cluster, the IAM entity user or role, such as a federated user that creates the cluster, is automatically granted system:masters permissions in the cluster's RBAC configuration. To grant additional AWS users or roles the ability to interact with your cluster, you must edit the aws-auth ConfigMap within Kubernetes.`

Reply:
We could access region with `aws.getRegion()` and define it explicitly, assuming that will pick up the parent provider for the api call.

Reply:
> That would be really weird. Is there not a way to just pass it the role you want it to associate with IAM access?

I too have not seen any explicit association between the role used at cluster creation time and the EKS cluster. Per the AWS quote, this is implicitly inferred.